### ISSUE TITLE
Potential oversight in accuracy metric

### ISSUE THREAD

**Issue by Space9999** [jsbaan/transformer-from-scratch#7]
https://github.com/jsbaan/transformer-from-scratch/blob/40078344c7a47a37ba7733b83b119e3bfb2d9079/train.py#L59-L71

You set the pad tokens to -100 to make pytorch ignore them in the loss metric. However, the accuracy does not exclude the padding positions in target batch from the calculation (it would be trying to compare -100 to whatever is set in the equivalent position). This could negatively impact the accuracy score.

A potential fix would be creating a mask that would check if any part of target batch is -100 and only comparing the predictions to the target batch that would match that mask.

**Comment by jsbaan** [comment]
Good catch, thanks! I fixed it in PR #8
